packages/alea-frontend/pages/api/get-slides.ts

```diff
@@ -8,38 +9,69 @@ import {
   getSectionSlides,
 } from '@stex-react/api';
 import { NextApiRequest, NextApiResponse } from 'next';
-
+interface SlidesWithCSS {
+  slides: Slide[];
+  css: CSS[];
+}
 async function recursivelyExpandSlideElementsExcludeSections(
   slideElems: SlideElement[],
-  sectionId: string
-): Promise<Slide[]> {
+  sectionId: string,
+  accumulatedCSS: CSS[] = []
+): Promise<{ slides: Slide[]; css: CSS[] }> {
   const elems: (Extract<SlideElement, { type: 'Paragraph' | 'Slide' }> | Slide)[] = [];
+  const cssSet = new Set<string>();
+  accumulatedCSS.forEach((css) => cssSet.add(JSON.stringify(css)));
   for (const slideElem of slideElems) {
     if (slideElem.type === 'Inputref') {
       const slidesData = await getSectionSlides(slideElem.uri);
       if (slidesData?.length > 0) {
         const [css, slideElems] = slidesData;
-        elems.push(...(await recursivelyExpandSlideElementsExcludeSections(slideElems, sectionId)));
+        if (css && Array.isArray(css)) {
```

Comment:
no need to check `Array.isArray(css)`. The flams api that we call guarnatees the type.
You can also skip the `if (css)` check. If you want to be sure, you can update the forEach to avoid potential crash when `css` is empty:
`(css??[]).forEach( (cssItem)...`